src/pool/cli.c

```diff
@@ -238,6 +276,41 @@ dc_pool_hdl_unlink(struct dc_pool *pool)
 	daos_hhash_link_delete(&pool->dp_hlink);
 }
 
+static void dc_pool_handle_cache_fini(void)
+{
+	struct dc_pool_handle_cache *dphc;
+	struct dc_pool_handle_cache *tmp;
+
+	D_SPIN_LOCK(&dc_pool_cache_lock);
```

Comment:
Lock is not needed when it comes here. A further question is what happens if there are still io flighting. but eg daos_fini() is called in one thread, it looks now we don't have some barrier in daos_fini() to wait any existing IO to finished.

Or that should be guaranteed by users?  @mchaarawi ?

Reply:
daos_fini has a refcount mechanism already to guard against this.
note that if the application only call daos_init() once from one thread, it is expected from the application to wait for all threads to complete to call daos_fini(). but it is fine if all threads call init/fini themselves.